spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/StreamTemplate.java

```diff
@@ -83,16 +75,15 @@ public StreamDefinitionResource createStream(String name, String definition, boo
 
 	@Override
 	public void deploy(String name, Map<String, String> properties) {
-		String uri = deploymentLink.expand(name).getHref();
+		String uri = deploymentsLink.getHref() + "/{name}";
```

Comment:
We should not be constructing URIs by hand. Something got wrong with the previous approach, **that** needs to be fixed). Are you saying e.g. the `streams/definitions/definition` relation was not correctly set (anymore?)